nan.h

```diff
@@ -1077,10 +1077,23 @@ class Utf8String {
         }
         const int flags =
             v8::String::NO_NULL_TERMINATION | imp::kReplaceInvalidUtf8;
-#if NODE_MAJOR_VERSION >= 10
-        length_ = string->WriteUtf8(v8::Isolate::GetCurrent(), str_, static_cast<int>(len), 0, flags);
-#else
+        // See https://github.com/nodejs/nan/issues/832.
+        // Disable the warning as there is no way around it.
+        // TODO(bnoordhuis) Use isolate-based version in Node.js v12.
```

Comment:
I think the isolate version of `WriteUtf8` is already available since v11 so `#if NODE_MAJOR_VERSION >= 11` should be fine here.

Reply:
What about
```c++
#if !NODE_VERSION_AT_LEAST(10,9,0)
```
It will use the isolate-based version for 10.10 and higher.

Reply:
@MaartenBent There are situations where the node version used to build differs from that one used to execute the program. Therefore if you build with 10.10.0 but execute with 10.9.0 it will not work as the new API is not there. This is not intended as usually a native addon binary works for all minors of a major version.

Reply:
You're right, I didn't consider ABI compatibility.

Reply:
> I think the isolate version of WriteUtf8 is already available since v11 so #if NODE_MAJOR_VERSION >= 11 should be fine here.

Right, that's true. Updated.